CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# How to contribute to JuMP
+
+Welcome! This document explains some of the ways you can contribute to JuMP.
+
+## Join the Discourse forum
+
+First up, join the [Discourse forum](https://discourse.julialang.org/c/domain/opt).
+
+The forum is a good place to ask questions about how to use JuMP. You can also
+use the forum to discuss possible feature requests and bugs before raising a
+Github issue (more on this below).
+
+Aside from asking questions, the easiest way you can contribute to JuMP is to
+help answer questions on Discourse!
+
+## Improve the documentation
+
+Chances are, if you asked (or answered) a question on Discourse, then it is
+usually a sign that the documentation could be improved. Moreover, since it is
+your question, you are probably the best-placed person to improve it!
+
+If your change is small (like fixing typos, or one or two sentence corrections),
+the easiest way to do this is via Github's online editor. (Github has
+[help](https://help.github.com/articles/editing-files-in-another-user-s-repository/)
+on how to do this.)
+
+If your change is larger, or touches multiple files, you will need to make the
+change locally and then use Git to submit a pull request. (See [Contribute code to JuMP](#Contribute code to JuMP)
+below for more on this.)
+
+If you need any help, come join our [Gitter](https://gitter.im/JuliaOpt/JuMP-dev)
+channel and we will walk you through the process.
+
+## Write examples
+
+Another way to contribute to JuMP is to add examples. The easiest way to do this
```

Comment:
The structure for examples is a bit up in the air now (https://github.com/JuliaOpt/JuMP.jl/issues/1175), so I'm not sure if we're ready to accept new ones at this point.